Makefile

```diff
@@ -13,13 +13,16 @@ OUT_BIN_NAME := DG
 OUT_BIN := $(OUT_DIR)/$(OUT_BIN_NAME)
 
 SRC_FILES := $(wildcard */*.cxx)
+# Define OBJ_FILES in case wanting to use in future
+OBJ_FILES := $(SRC_FILES:.cxx=.o)
 
 # .PHONY:
 
 # TODO automatic dep file creation (to track included files) by using cpp
 
-# TODO add further dependencies, object files instead of sources
```

Comment:
Remove because not compiling to object files and linking separately. To my knowledge, Make can handle C object files without a need for writing rules for these object files. Not sure if the support is there for C++ and Clang (at least uses same option (-c) for compiling to object files, so could be possible).